sync_check_terraform/main.tf

```diff
@@ -0,0 +1,178 @@
+  }
+}
+
+locals {
+  name                  = "forest-sync-check" # droplet name
+  size                  = "so-2vcpu-16gb"     # droplet size
+  image                 = "fedora-36-x64"
+  slack_channel         = "#forest-notifications" # slack channel for notifications
+  region                = "fra1"
+  source_addresses      = ["0.0.0.0/0", "::/0"]
+  destination_addresses = ["0.0.0.0/0", "::/0"]
+  project               = "Forest-DEV"
+
+  # Variable passthrough:
+  slack_token        = var.slack_token
+  digitalocean_token = var.do_token
+}
+
+terraform {
+  required_version = "~> 1.3"
+
+  required_providers {
+    digitalocean = {
+      source  = "digitalocean/digitalocean"
+      version = "~> 2.0"
+    }
+  }
+}
+
+provider "digitalocean" {
+  token = var.do_token
+}
+
+// Ugly hack because 'archive_file' cannot mix files and folders.
+data "external" "sources_tar" {
+  program = ["sh", "${path.module}/prep_sources.sh", "${path.module}"]
+}
+
+data "local_file" "sources" {
+  filename = data.external.sources_tar.result.path
+}
+
+// Note: The init.sh file is also included in the sources.zip such that the hash
+// of the archive captures the entire state of the machine.
+data "local_file" "init" {
+  filename = "${path.module}/service/init.sh"
+}
+
+data "digitalocean_ssh_keys" "keys" {
+  sort {
+    key       = "name"
+    direction = "asc"
+  }
+}
+
+resource "digitalocean_droplet" "forest" {
+  image  = local.image
+  name   = local.name
+  region = local.region
+  size   = local.size
+  # Re-initialize resource if this hash changes:
+  user_data = data.local_file.sources.content_sha256
+  tags      = ["iac"]
+  ssh_keys  = data.digitalocean_ssh_keys.keys.ssh_keys.*.fingerprint
+
+  graceful_shutdown = false
+
+  connection {
+    host = self.ipv4_address
+    user = "root"
+    type = "ssh"
+  }
+
+  # Push the sources.tar file to the newly booted droplet
+  provisioner "file" {
+    source      = data.local_file.sources.filename
+    destination = "/root/sources.tar"
+  }
+
+  provisioner "remote-exec" {
+    inline = [
+      "tar xf sources.tar",
+      # Set required environment variables
+      "echo 'export FOREST_TAG=edge' >> .forest_env",
+      "echo 'export FOREST_TARGET_DATA=/volumes/forest_data' >> .forest_env",
+      "echo 'export FOREST_TARGET_SCRIPTS=/volumes/sync_check' >> .forest_env",
+      "echo 'export FOREST_TARGET_RUBY_COMMON=/volumes/ruby_common' >> .forest_env",
+      "echo 'export FOREST_SLACK_API_TOKEN=\"${var.slack_token}\"' >> .forest_env",
+      "echo 'export FOREST_SLACK_NOTIF_CHANNEL=\"${local.slack_channel}\"' >> .forest_env",
+      "echo 'source .forest_env' >> .bashrc",
+      "source ~/.forest_env",
+      "/bin/bash ./init.sh > init_log.txt",
+      "nohup /bin/bash ./run_service.sh > run_service_log.txt &",
+      "cp ./restart.service /etc/systemd/system/",
+      "systemctl enable restart.service",
+      # Exiting without a sleep sometimes kills the script :-/
```

Comment:
Do we know why? I mean, I'm okay with the workaround, but it'd be nice to know the reason.